### ISSUE TITLE
[raysgd] Implement callbacks

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#6830]
### Describe your feature request

The current API doesn't make it possible to implement learning rate schedulers (https://pytorch.org/docs/stable/optim.html). We can provide this functionality with stateful callbacks, like how Keras handles this.
